src/Data/BitVector/Sized/Internal.hs

```diff
@@ -1,419 +1,744 @@
+  where rot = rot' `mod` wNatural
+        rightChunk = lshr bv rot
+        leftChunk = shl w bv (wNatural - rot)
+        wNatural = natValue w
+
+-- | The zero bitvector of any width.
+zero :: NatRepr w -> BV w
+zero w = checkNatRepr w $ BV 0
+
+-- | The bitvector with value 1, of any positive width.
+one :: 1 <= w => NatRepr w -> BV w
+one w = checkNatRepr w $ BV 1
+
+-- | The bitvector whose value is its own width, of any width.
+width :: NatRepr w -> BV w
+width w = checkNatRepr w $ BV (intValue w)
+
+-- | The 'BV' that has a particular bit set, and is 0 everywhere
+-- else.
+bit :: ix+1 <= w
+    => NatRepr w
+    -- ^ Desired output width
+    -> NatRepr ix
+    -- ^ Index of bit to set
+    -> BV w
+bit w ix = checkNatRepr w $ BV (B.bit (widthVal ix))
+
+-- | Like 'bit', but without the requirement that the index bit refers
+-- to an actual bit in the output 'BV'. If it is out of range, just
+-- silently return the zero bitvector.
+bit' :: NatRepr w
+     -- ^ Desired output width
+     -> Natural
+     -- ^ Index of bit to set
+     -> BV w
+bit' w ix = mkBV w (B.bit (fromIntegral ix))
+
+-- | @setBit w bv ix@ is the same as @or bv (bit w ix)@.
+setBit :: ix+1 <= w
+       => NatRepr ix
+       -- ^ Index of bit to set
+       -> BV w
+       -- ^ Original bitvector
+       -> BV w
+setBit ix bv = or bv (BV (B.bit (widthVal ix)))
+
+-- | Like 'setBit', but without the requirement that the index bit
+-- refers to an actual bit in the input 'BV'. If it is out of range,
+-- just silently return the original input.
+setBit' :: NatRepr w
+        -- ^ Desired output width
+        -> Natural
+        -- ^ Index of bit to set
+        -> BV w
+        -- ^ Original bitvector
+        -> BV w
+setBit' w ix bv = or bv (mkBV' w (B.bit (fromIntegral ix)))
+
+-- | @clearBit w bv ix@ is the same as @and bv (complement (bit w ix))@.
+clearBit :: ix+1 <= w
+         => NatRepr w
+         -- ^ Desired output width
+         -> NatRepr ix
+         -- ^ Index of bit to clear
+         -> BV w
+         -- ^ Original bitvector
+         -> BV w
+clearBit w ix bv = and bv (complement w (BV (B.bit (widthVal ix))))
+
+-- | Like 'clearBit', but without the requirement that the index bit
+-- refers to an actual bit in the input 'BV'. If it is out of range,
+-- just silently return the original input.
+clearBit' :: NatRepr w
+          -- ^ Desired output width
+          -> Natural
+          -- ^ Index of bit to clear
+          -> BV w
+          -- ^ Original bitvector
+          -> BV w
+clearBit' w ix bv = and bv (complement w (mkBV' w (B.bit (fromIntegral ix))))
+
+-- | @complementBit w bv ix@ is the same as @xor bv (bit w ix)@.
+complementBit :: ix+1 <= w
+              => NatRepr w
+              -- ^ Desired output width
+              -> NatRepr ix
+              -- ^ Index of bit to flip
+              -> BV w
+              -- ^ Original bitvector
+              -> BV w
+complementBit _w ix bv = xor bv (BV (B.bit (widthVal ix)))
```

Comment:
Related to another comment, this width repr is also unused (and not required for the return type)

Reply:
Good catch, thanks

Reply:
Thanks